### PR TITLE
Bug 1217860 - beta build

### PR DESCRIPTION
**Pull request by fluffyemily**
https://bugzilla.mozilla.org/show_bug.cgi?id=1217860

### CODE REVIEW

File: fastlane/BaseFastfile [hunk 1]
```diff
@@ -151,6 +142,66 @@ platform :ios do
     enterprise_build(options: options)
   end
 
+  desc "Builds an beta build for the Firefox channel. Takes the following arguments: \n \
```
Comment (sleroux) [inline fastlane/BaseFastfile:145]: Nit: an -> a

File: fastlane/BaseFastfile [hunk 2]
```diff
@@ -151,6 +142,66 @@ platform :ios do
...
+  :build            the build number for this build. Defaults to current build number \n \
+  :version          the version number for this build. Defaults to current version number \n \
+  :branch           the branch to build (will create if doesn't already exist. A new branch with a predictable username will \
+    be created if this is not provided) \n \
+  :base_branch      the branch with off of which the build should be based (current branch if empty) \n \
+  :build_name       the name of the resultant build. Defaults to firefox-ios-l10n-<:version>-<:build>-<timestamp> \n \
+  :username         the Apple ID to use to log into the developer portal for SIGH \n \
+  :localise         if true, the branch will be localised. If absent or false no localisation will occur \n \
+  :changelog        the path to the changelog for the beta build"
+  lane :beta do |options|
```
Comment (sleroux) [inline fastlane/BaseFastfile]: This lane should be named :fennec since :beta is already defined above.

**Comment by sleroux** [comment]
Got it to complete the build and archive but I get the following error after it zips up the symbols:

`﻿Could not find action or lane 'params'`

I think there might be a misnamed method being called somewhere in the lane?

**Comment by sleroux** [comment]
Hmm possibily this line? 

https://github.com/mozilla/firefox-ios-build-tools/pull/10/files#diff-d3de54919c7775fe62204a85ee939ee5R176

**Comment by fluffyemily** [comment]
It should provide a link to a log file with the error in it. Can you add it?

**Comment by sleroux** [comment]
https://gist.github.com/sleroux/f82124142d81e59cd523

**Comment by sleroux** [comment]
I found this link that might help with our git/l10n repo addition woes:
`Do a “git export” (like “svn export”)?`
http://stackoverflow.com/questions/160608/do-a-git-export-like-svn-export

**Comment by sleroux** [comment]
I ran into an issue with the localise.sh script:
https://gist.github.com/sleroux/64f04b667e40e50147ed

Orignally it failed because SVN didn't recognize the server certificate. After manually accepting the cert, I was able to export it but then ran into the issue above.

**Comment by sleroux** [comment]
Ran into another issue this morning. Did a fresh clone, updated the fastfile to point to this branch, commited, and tried running the beta lane:

```

Client (en): Client/en.lproj/InfoPlist.strings
Deactivating virtualenv
Committing localised files
On branch firefox-2.0-2-20160121T145055Z
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

    modified:   Client.xcodeproj/project.pbxproj

Untracked files:
  (use "git add <file>..." to include in what will be committed)

    firefox-ios-l10n/

no changes added to commit (use "git add" and/or "git commit -a")
fatal: pathspec 'Extensions/*/*.lproj' did not match any files

```

**Comment by fluffyemily** [comment]
what command did you run?

**Comment by sleroux** [comment]
`fastlane beta version:2.0 build:2 username:xxx localise:true`

I notice there isn't any Extension/_/_.lproj to be found. Do we need this as part of the checkin?

**Pull request merged**